python/grass/script/imagery.py

```diff
@@ -0,0 +1,155 @@
+    The default is to use semantic_labels. Note that map metadata
+    of the maps in the group have to be read to get the semantic label,
+    in addition to the group file. The same metadata is read when the
+    "metadata" is requested as dict_values. Other supported dict_values
+    are "map_names" (default), "semantic_labels", or "indices".
+
+    The function can also operate on the level of subgroups. In case a
+    non-existing (or empty sub-group) is requested a warning is printed
+    and an empty dictionary is returned (following the behavior of i.group).
+
+    Example::
+
+    >>> run_command("g.copy", raster="lsat7_2000_10,lsat7_2000_10")
+    >>> run_command("r.support", raster="lsat7_2000_10", semantic_label="L8_1")
+    >>> run_command("g.copy", raster="lsat7_2000_20,lsat7_2000_20")
+    >>> run_command("r.support", raster="lsat7_2000_20", semantic_label="L8_2")
+    >>> run_command("g.copy", raster="lsat7_2000_30,lsat7_2000_30")
+    >>> run_command("r.support", raster="lsat7_2000_30", semantic_label="L8_3")
+    >>> run_command("i.group", group="L8_group",
+    >>>             input="lsat7_2000_10,lsat7_2000_20,lsat7_2000_30")
+    >>> group_to_dict("L8_group")  # doctest: +ELLIPSIS
+    {"L8_1": "lsat7_2000_10", ... "L8_3": "lsat7_2000_30"}
+    >>> run_command("g.remove", flags="f", type="group", name="L8_group")
+    >>> run_command("g.remove", flags="f", type="raster",
+    >>>             name="lsat7_2000_10,lsat7_2000_20,lsat7_2000_30")
+
+    :param str imagery_group_name: Name of the imagery group to process (or None)
+    :param str subgroup: Name of the imagery sub-group to process (or None)
+    :param str dict_keys: What to use as key for dictionary. Can bei either
+                         "semantic_labels" (default), "map_names" or "indices"
+    :param str dict_values: What to use as values for dictionary. Can bei either
+                           "map_names" (default), "semanic_labels", "indices" or
+                           "metadata" (to return dictionaries with full map metadata)
+    :param bool fill_semantic_label: If maps in a group do not have a semantic
+                                     label, their index in the group is used
+                                     instead (default). Otherwise None / "none"
+                                     is used.
+    :param dict env: Environment to use when parsing the imagery group
+
+    :return: dictionary representing an imagery group with it's maps and their
+             semantic labels, row indices in the group, or metadata
+    :rtype: dict
+    """
+    group_dict = {}
+    try:
+        maps_in_group = (
+            read_command(
+                "i.group",
+                group=imagery_group_name,
+                subgroup=subgroup,
+                flags="g",
+                quiet=True,
+                env=env,
+            )
+            .strip()
+            .split()
+        )
+    except CalledModuleError as error:
+        raise error
```

Comment:
If you are not doing anything with the error, you can just not use try-except. This would be good to raise a different type of exception with raise-from.

Reply:
So, you mean dropping try-except and just running the module and rely on the error message there?

Reply:
I removed the ty-except now...

Reply:
Removing is fine with me. Just to be clear, if you do catch in except and raise you do almost the same traceback as if there is no raise. except + raise would make sense if you would be adding more info and/or change the type of the exception.